File: rps/rps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleRps
{
    class Rps
    {
        static string[] answers = new string[] { "rock", "paper", "scissors" };
        static Dictionary<string, Dictionary<string, int>> results = new Dictionary<string, Dictionary<string, int>>() {
            {"rock", new Dictionary<string, int>() {
                { "rock", 0 }, { "paper", -1 }, { "scissors", 1 }
            }},
            { "paper", new Dictionary<string, int>() {
                { "rock", 1 }, { "paper", 0 }, { "scissors", -1 }
            }},
            { "scissors", new Dictionary<string, int>() {
                { "rock", -1 }, { "paper", 1 }, { "scissors", 0 }
            }}
        };
        static Dictionary<int, string> resultTexts = new Dictionary<int, string>() {
            {-1, "You lose!"},
            {0, "It's a tie!"},
            {1, "You win!"}
        };
        static int wins = 0;
        static int losses = 0;
        static bool isPlaying = true;

        static void Main(string[] args)
        {
            Console.WriteLine("Rock, paper, scissors; best of five!");
            Console.WriteLine("Choose an action[rock, paper, scissors]");
            Random rnd = new Random();
            // Main game loop, to be broken only when game is finished.
            while (isPlaying)
            {
                String action = Console.ReadLine();
                if (answers.Contains(action))
                {
                    int cpu = rnd.Next(0, 3);
                    string cpuAction = answers[cpu];
                    Console.WriteLine("The CPU chose {0}", cpuAction);
                    int result = results[action][cpuAction];
                    if (result == 1)
                    {
                        wins++;
                    }
                    else if (result == -1)
                    {
                        losses++;
                    }
                    Console.WriteLine(resultTexts[result] + " ({0} - {1})", wins, losses);
                    if (wins > 2 || losses > 2)
                    {
                        if (wins > losses)
                        {
                            Console.WriteLine("You won the game!");
                        }
                        else
                        {
                            Console.WriteLine("You lost the game...");
                        }
                        isPlaying = false;
                    }
                    else
                    {
                        Console.WriteLine("Choose a new action.");
                    }
                }
                else
                {
                    Console.WriteLine("Wrong action!");
                }
            }
            // Out of game loop, add a new ReadLine so the player can read the result.
            Console.ReadLine();
        }
    }
}

File: rps/js/rps.js
'use strict';

let readline = require('readline');
let rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

function color (color) {
	let colors = {
		'black': '30m',
		'blue': '34m',
		'green': '32m',
		'cyan': '36m',
		'red': '31m',
		'purple': '35m',
		'brown': '33m',
		'lightgray': '37m',
		'darkgray': '1;30m',
		'lightblue': '1;34m',
		'lightgreen': '1;32m',
		'lightcyan': '1;36m',
		'lightred': '1;31m',
		'lightpurple': '1;35m',
		'yellow': '1;33m',
		'white': '1;37m',
		'none': '0m'
	};
	let c = (colors[color]) ? colors[color] : '0m';
	return "\x1B[" + c;
}

let wins = 0;
let losses = 0;
let actions = ['rock', 'paper', 'scissors'];
let results = {
	'rock': {'rock': 0, 'paper': -1, 'scissors': 1},
	'paper': {'rock': 1, 'paper': 0, 'scissors': -1},
	'scissors': {'rock': -1, 'paper': 1, 'scissors': 0}
};
let resultText = {'-1': color('red') + 'You lose!', '0': color('purple') + "It's a tie!", '1': color('green') + 'You win!'};
let onGoing = true;
let finished = false;

console.log(color('white') + 'Rock, paper, scissors; best of five!' + color());
console.log('Choose an action [rock, paper, scissors]');

rl.on('line', function (answer) {
	if (actions.indexOf(answer) !== -1) {
		let cpu = actions[Math.floor(Math.random() * 3)];
		console.log('The CPU chose ' + cpu);
		let result = results[answer][cpu];
		if (result === 1) wins++;
		else if (result === -1) losses++;
		console.log(resultText[result] + color() + ' (' + wins + ' - ' + losses + ')');
		if (wins > 2 || losses > 2) {
			if (wins > losses) {
				console.log(color('lightgreen') + 'You won the game!!' + color());
			} else {
				console.log(color('lightred') + 'You lost the game...' + color());
			}
			rl.close();
		} else {
			console.log('Choose a new action.');
		}
	} else {
		console.log(color('red') + 'Wrong action!' + color());
	}
});
